src/server/pps/server/api_server.go

```diff
@@ -2662,10 +2662,15 @@ func (a *apiServer) getPachClient() *client.APIClient {
 			panic(fmt.Sprintf("pps failed to initialize pach client: %v", err))
 		}
 		// Initialize spec repo
-		if err := a.pachClient.CreateRepo(ppsconsts.SpecRepo); err != nil {
-			if !isAlreadyExistsErr(err) {
-				panic(fmt.Sprintf("could not create pipeline spec repo: %v", err))
+		if err := a.sudo(a.pachClient, func(superUserClient *client.APIClient) error {
```

Comment:
It seems like we're covering an edge case here, but I'm not sure exactly what. Is it that before the spec repo would've used non su creds to create the internal spec repo? Which could fail if ... ?

Is it worth writing a test case for this?

Reply:
The change here is actually sort of minor. Basically, this code tries to create a spec repo, and if it already exists, this code ignores the error and continues.

After this PR, if you try to create a repo without logging in, and the repo also already exists, you get `auth token not found`, which breaks this code even though the repo also already exists.

Wrapping this in sudo guarantees that if pachd starts after auth has been activated (which pretty much means that the spec repo has already been created, though there are no hard checks in the auth server that the spec repo exists) then pachd gets a `__spec__ already exists` error here and happily ignores it, rather than an `auth token not found` error which breaks here.

I think this code is actually tested by virtue of the fact that there are auth tests that start pipelines, and each pipeline starts a sidecar which tries to create the spec repo. If this weren't wrapped in sudo, every sidecar that started after auth was activated would crash.

Reply:
Just confirmed, this is covered by existing auth tests (when I revert this delta, they fail)